libs/execution_context/ebpf_maps.c

```diff
@@ -1770,6 +1770,30 @@ ebpf_map_create(
     uint32_t cpu_count;
     cpu_count = ebpf_get_cpu_count();
     ebpf_map_definition_in_memory_t local_map_definition = *ebpf_map_definition;
+    if (ebpf_map_definition->key_size == 0) {
+        switch (local_map_definition.type) {
```

Comment:
Rather than switch statements line 1774, 1785, and 1797, it may be more maintainable to add fields to `ebpf_map_function_table_t` to test here.  Otherwise my fear is that an entry will be added there and updating this function will be missed.